pkg/operators/crdmanager/manager.go

```diff
@@ -0,0 +1,216 @@
+	// remove version label if skip clean
+	if skip {
+		existing, err := m.client.Get(ctx, name, metav1.GetOptions{})
+		switch {
+		case apierrors.IsNotFound(err):
+			return nil
+		case err != nil:
+			return err
+		}
+		accessor, err := meta.Accessor(existing)
+		if err != nil {
+			return err
+		}
+		labels := accessor.GetLabels()
+		if _, ok := labels[versionLabelKey]; ok {
+			delete(labels, versionLabelKey)
+		}
+		accessor.SetLabels(labels)
+		_, err = m.client.Update(ctx, existing, metav1.UpdateOptions{})
+		return err
+	}
+
+	err := m.client.Delete(ctx, name, metav1.DeleteOptions{})
+	switch {
+	case apierrors.IsNotFound(err):
+		return nil
+	case err == nil:
+		return &RemainingCRDError{RemainingCRDs: []string{name}}
+	}
+
+	return err
+}
+
+func (m *Manager[T]) Clean(ctx context.Context, skip bool, manifests resourceapply.AssetFunc, files ...string) error {
+	var errs []error
+	var remainingCRDs []string
+
+	for _, file := range files {
+		objBytes, err := manifests(file)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		requiredObj, _, err := genericCodec.Decode(objBytes, nil, nil)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		accessor, err := meta.Accessor(requiredObj)
+		if err != nil {
+			return err
+		}
+
+		err = m.CleanOne(ctx, accessor.GetName(), skip)
+		var remainingErr *RemainingCRDError
+		switch {
+		case errors.As(err, &remainingErr):
+			remainingCRDs = append(remainingCRDs, accessor.GetName())
+		case err != nil:
+			errs = append(errs, err)
+		}
+	}
+
+	if len(errs) > 0 {
+		return utilerrors.NewAggregate(errs)
+	}
+	if len(remainingCRDs) > 0 {
+		return &RemainingCRDError{RemainingCRDs: remainingCRDs}
+	}
+
+	return nil
+}
+
+func (m *Manager[T]) Apply(ctx context.Context, manifests resourceapply.AssetFunc, files ...string) error {
+	var errs []error
+
+	for _, file := range files {
+		objBytes, err := manifests(file)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		requiredObj, _, err := genericCodec.Decode(objBytes, nil, nil)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		err = m.applyOne(ctx, requiredObj.(T))
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	return utilerrors.NewAggregate(errs)
+}
+
+func (m *Manager[T]) applyOne(ctx context.Context, required T) error {
+	accessor, err := meta.Accessor(required)
+	if err != nil {
+		return err
+	}
+	existing, err := m.client.Get(ctx, accessor.GetName(), metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		_, err := m.client.Create(ctx, required, metav1.CreateOptions{})
+		klog.Infof("crd %s is created", accessor.GetName())
+		return err
+	}
+	if err != nil {
+		return err
+	}
+
+	// if existingVersion is higher than the required version, do not update crd.
+	accessor, err = meta.Accessor(existing)
+	if err != nil {
+		return err
+	}
+	existingVersion, ok := accessor.GetLabels()[versionLabelKey]
+	requiredVersion := version.Get().GitVersion
```

Comment:
If we are in development phase, the `version.Get().GitVersion` could be like `v0.9.0-16-g889bd8b`. can this value be comparable?